src/adapter/src/catalog/open.rs

```diff
@@ -755,6 +757,31 @@ impl Catalog {
 
             let mut state = Catalog::load_catalog_items(&mut txn, &state)?;
 
+            // Run a one-time migration to clear the existing sources.
+            //
+            // TODO(parkmycar): Remove after v0.92.
+            const ACTIVITY_LOG_ONE_TIME_MIGRATION: &str = "activity_log_v_0_92_migration";
+            let ran_migration = txn.get_config(ACTIVITY_LOG_ONE_TIME_MIGRATION.to_string())?.is_some();
+            if !ran_migration {
+                let candidates = [
+                    MZ_PREPARED_STATEMENT_HISTORY.name,
+                    MZ_SQL_TEXT.name,
+                    MZ_STATEMENT_LIFECYCLE_HISTORY.name,
+                    MZ_STATEMENT_EXECUTION_HISTORY.name,
+                ];
+                let to_migrate: Vec<_> = txn
+                    .get_system_items()
+                    .filter(|mapping| {
+                        mapping.description.schema_name == MZ_INTERNAL_SCHEMA && candidates.contains(&mapping.description.object_name.as_str())
+                    })
+                    .map(|mapping| mapping.unique_identifier.id)
+                    .collect();
+                tracing::info!(?to_migrate, "running one time migration");
+                migrated_builtins.extend(to_migrate);
+
+                txn.set_config(ACTIVITY_LOG_ONE_TIME_MIGRATION.to_string(), Some(1))?;
```

Comment:
What happens if we crash after committing `txn` but before actually migrating the tables?

Reply:
As part of the transaction we run a DropSource command, which should remove all capabilities for that source. So if we crash after committing the transaction but before migrating the tables, we'll leak the underlying shard but that should get detected as part of the startup